src/test/compile-fail/pub-struct-field-span.rs

```diff
@@ -18,6 +20,11 @@ struct Foo {
     //~^ error: field `bar` is already declared [E0124]
     bar: u8,
 
+    pub
```

Comment:
May I suggest you use a `ui` test? That way you can actually directly record the highlight that is expected, giving more precision.